bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlProvider.java

```diff
@@ -700,13 +701,38 @@ private void validateNonPrimitiveArray(TomlNode tomlValue, String variableName,
                 tomlValue = getValueFromKeyValueNode(tomlValue);
                 validateUnionValueArray(tomlValue, variableName, arrayType, (BUnionType) elementType);
                 break;
+            case TypeTags.TABLE_TAG:
+                validateTableValueArray(tomlValue, variableName, arrayType, (BTableType) elementType);
+                break;
             default:
                 Type effectiveType = ((IntersectionType) elementType).getEffectiveType();
                 validateNonPrimitiveArray(tomlValue, variableName, arrayType, effectiveType);
                 break;
         }
     }
 
+    private void validateTableValueArray(TomlNode tomlValue, String variableName, ArrayType arrayType,
+                                         BTableType elementType) {
+        TomlType tomlType = tomlValue.kind();
+        switch (tomlType) {
+            case ARRAY:
+                visitedNodes.add(tomlValue);
+                List<TomlValueNode> nodeList = ((TomlArrayValueNode) tomlValue).elements();
+                validateArraySize(tomlValue, variableName, arrayType, nodeList.size());
+                for (int i = 0; i < nodeList.size(); i++) {
+                    validateValue(nodeList.get(i), variableName + "[" + i + "]", elementType);
+                }
+                return;
+            case KEY_VALUE:
+                visitedNodes.add(tomlValue);
+                tomlValue = ((TomlKeyValueNode) tomlValue).value();
+                validateTableValueArray(tomlValue, variableName, arrayType, elementType);
+                return;
+            default:
+                throwTypeIncompatibleError(tomlValue, variableName, arrayType);
```

Comment:
Can we adda test for this?

Reply:
I added unit tests for this case. 
https://github.com/ballerina-platform/ballerina-lang/pull/37458/files/5d54cacacbb67ccf9dd27acf91c0c783bcf0e7de#diff-be31eb2ce988265fc1a0d737443b5f246106ca8fb75ef3929434887d2a85fae0

As the `throwTypeIncompatibleError` method call throws an exception, this line shows as not covered.